### PR TITLE
Repeat count required

### PR DESCRIPTION
**Pull request by emord**
http://manage.dimagi.com/default.asp?183425#1022284

@phillipm 

cc: @TylerSheffels

### CODE REVIEW

**Comment by phillipm** [comment]
Context: This is the real fix for http://manage.dimagi.com/default.asp?182818 contrary to the previous fix attempt, https://github.com/dimagi/Vellum/pull/537, which didn't discriminate based on whether the repeat was or was not in a field list.

This looks right, though I'll let someone else review & merge since I'm not up on my javascript.

**Comment by TylerSheffels** [comment]
:+1: after tests

File: src/mugs.js [hunk 1]
```diff
@@ -1538,8 +1538,19 @@ define([
...
-                presence: 'required',
-                widget: widgets.droppableText
+                presence: 'optional',
+                widget: widgets.droppableText,
+                validationFunc: function (mug) {
+                    var parentMug = mug.parentMug,
+                        value = $.trim(mug.p.repeat_count);
+
+                    if (!value &&
+                        parentMug && parentMug.__className === 'FieldList') {
```
Comment (orangejenny) [inline src/mugs.js]: isn't it potentially any ancestor, not only the direct parent?

Comment (phillipm) [inline src/mugs.js]: sounds right, good catch!

Comment (emord) [inline src/mugs.js]: https://github.com/dimagi/Vellum/commit/87fca042da5475ea6987d77a64d3915cadc32ebe

**Pull request merged**